dashboard/src/app/ide/ide.controller.ts

```diff
@@ -9,39 +9,57 @@
  *   Codenvy, S.A. - initial API and implementation
  */
 'use strict';
+import IdeSvc from './ide.service';
+import IdeIFrameSvc from './ide-iframe/ide-iframe.service';
+import {RouteHistory} from '../../components/routing/route-history.service';
+import {CheWorkspace} from '../../components/api/che-workspace.factory';
 
 /**
  * This class is handling the controller for the IDE
  * @author Florent Benoit
  */
 class IdeCtrl {
+  $rootScope: ng.IRootScopeService;
+  $timeout: ng.ITimeoutService;
+  ideSvc: IdeSvc;
+  ideIFrameSvc: IdeIFrameSvc;
+  cheWorkspace: CheWorkspace;
+
+  hasData: boolean;
+  workspaces: any[];
+  selectedWorkspace: any = null;
+  selectedWorkspaceExists: boolean = true;
+  selectedWorkspaceName: string = null;
 
   /**
    * Default constructor that is using resource
    * @ngInject for Dependency injection
    */
-  constructor(ideSvc, $routeParams, ideIFrameSvc, $rootScope, cheWorkspace, $timeout, $location, routeHistory) {
+  constructor($location: ng.ILocationService, $rootScope: ng.IRootScopeService,
+              $routeParams: ng.route.IRouteParamsService, $timeout: ng.ITimeoutService, ideSvc: IdeSvc,
+              ideIFrameSvc: IdeIFrameSvc, cheWorkspace: CheWorkspace, routeHistory: RouteHistory) {
     this.ideSvc = ideSvc;
     this.ideIFrameSvc = ideIFrameSvc;
     this.$rootScope = $rootScope;
     this.cheWorkspace = cheWorkspace;
     this.$timeout = $timeout;
-    this.selectedWorkspace = null;
-    this.$rootScope.showIDE = false;
 
-    $rootScope.wantTokeepLoader = true;
+    (this.$rootScope as any).showIDE = false;
+    (this.$rootScope as any).wantTokeepLoader = true;
+
+    this.selectedWorkspaceExists = true;
 
     // search the selected workspace
-    let namespace = $routeParams.namespace;
-    let workspace = $routeParams.workspaceName;
+    let namespace = ($routeParams as any).namespace;
```

Comment:
ok it's because I have the feeling that ($routeParams as any) adds complexity when reading code. and because typescript definition says that we have an array not an object.